internal/wrappers/microsastengine/scans/micro-sast-grpc.go

```diff
@@ -0,0 +1,106 @@
+package scans
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	"github.com/checkmarx/ast-cli/internal/logger"
+	"github.com/google/uuid"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	healthpb "google.golang.org/grpc/health/grpc_health_v1"
+)
+
+type MicroSastWrapper struct {
+	port int
+}
+
+func NewMicroSastWrapper(port int) *MicroSastWrapper {
+	return &MicroSastWrapper{
+		port: port,
+	}
+}
+
+func (s *MicroSastWrapper) Scan(filePath string) (*ScanResult, error) {
+	options := []grpc.DialOption{
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+		grpc.WithBlock(),
+	}
+	data, err := os.ReadFile(filePath)
+	if err != nil {
+		fmt.Print(err)
+	}
+	localhostAddress := fmt.Sprintf("0.0.0.0:%d", s.port)
+	conn, err := grpc.NewClient(localhostAddress, options...)
+
+	defer func(conn *grpc.ClientConn) {
+		_ = conn.Close()
+	}(conn)
+
+	if err != nil {
+		logger.Printf("grpc.NewClient(%q): %v", localhostAddress, err)
```

Comment:
Maybe grpc.NewClient can be replaced with a more indicative log? maybe something like
"Error occurred while creating the grpc client in address %q. error: %v"

Reply:
and maybe also return the error as you di in line 85?